[docs] Add resource validation statuses description (#4544)
vividus-framework/vividus: docs/modules/plugins/pages/plugin-web-app-to-rest-api.adoc

.. If the status code is one of https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404[404 Not Found], https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405[405 Method Not Allowed], https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501[501 Not Implemented], https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503[503 Service Unavailable] then GET request will be executed
.. If the https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET[GET] status code is https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200[200 OK] then the resource validation is considered as passed, otherwise failed

Resource validation statuses:

[cols="1,4", options="header"]
|===
|Status
|Description

|`FAILED`

> |A resource validation is considered failed if an HTTP request to the resource returns a status code other than https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200[200 OK].

Reviewer: ```suggestion
|An HTTP request to the resource returns a status code other than https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200[200 OK].
```

|`BROKEN`

> a|A resource validation is considered broken if:

Reviewer: ```suggestion
a|Reasons:
```

> * the request to the page under test returns an empty response body;

Reviewer: ```suggestion
* the request to the page under test returns an empty HTTP response body;
```

> * the request to the page under test returns an empty response body;

Reviewer: keep consistency: `An HTTP request` vs `the request`

* the request to the page under test results in unexpected error;
* the relative page URL can not be resolved because the `web-application.main-page-url` property is not set;
* the resource has invalid URL format;
* the resource is missing `href` or `src` attributes;

> * the resource has `href` or `src` attributes but the URL format is invalid;

Reviewer: ```suggestion
* the resource has `href` or `src` attribute but its value is not a valid URL;
```

* the resource is a jump link that points to non-existent jump target.

|`PASSED`

> |A resource validation is considered passed if an HTTP request to the resource returns https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200[200 OK] status code.

Reviewer: ```suggestion
|An HTTP request to the resource returns https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200[200 OK] status code.
```

|`FILTERED`

> a|A resource validation is considered passed if:

Reviewer: ```suggestion
a|Reasons:
```

* the resource path matches the patterns specified by the `resource-checker.uri-to-ignore-regex` property;
* the resource path is equal to `#` (anchor);
* the resource is not a HTTP(S) resource.

|`SKIPPED`

> |A resource validation is considered skipped if it has already been performed, i.e. if the same resource might be present on several pages so we do not need to validate it twice.

Reviewer: ```suggestion
|A resource validation has already been performed, i.e. if the same resource might be present on several pages so we do not need to validate it twice.
```
